versioned/persist/adapter/src/test/java/org/projectnessie/versioned/persist/adapter/spi/TestFetchValuesUsingOpenAddressing.java

```diff
@@ -301,28 +402,29 @@ private static KeyListEntity getKeyListEntity(int segmentIndex, int entryCount)
         .build();
   }
 
-  private static ImmutableKeyListEntry getEmbeddedEntry(int bucketIndex) {
-    return getKeyListEntry("embedded_" + bucketIndex);
-  }
-
-  private static ImmutableKeyListEntry getKeyListEntry(String keyString) {
-    Key k = new KeyWithMaxIntHash(ImmutableList.of(keyString));
+  private static ImmutableKeyListEntry getKeyListEntry(Key k) {
     return ImmutableKeyListEntry.builder()
         .key(k)
         .payload((byte) 99)
-        .contentId(ImmutableContentId.of(keyString + "_id"))
+        .contentId(ImmutableContentId.of(k + "_id"))
         .build();
   }
 
   /** All instances return {@linkplain Integer#MAX_VALUE} from {@linkplain #hashCode()}. */
-  private static class KeyWithMaxIntHash extends Key {
+  private static class KeyWithSpecificHash extends Key {
```

Comment:
Before simplifying to `KeyWithMaxIntHash` in the final squash and merge, I had something similar to this, which got me wondering at the time about accidental equals-hashCode violation.  I don't think there's any such violation here now; it looks good.

The scope might be sufficiently restricted that this remains safe over the life of the test, with whatever edits might still lie in the future. On the other hand, if you wanted, you could maybe add an integrity check to the `equals` implementation, e.g. instead of

```
return Objects.equals(elements, that.getElements());
```

perhaps

```
boolean elementsEqual = Objects.equals(elements, that.getElements());
if (elementsEqual && hashCode != that.hashCode()) {
  throw new IllegalStateException(String.format(
    "Equal key instances (path %s) found with different hashcodes: %s vs %s",
    Joiner.on(".").join(elements), hashCode, that.hashCode()));
}
return elementsEqual;
```

This could just as well be a precondition on fewer lines.  I used an explicit if-throw just to usually avoid the Joiner.  I can't properly contextualize this as a line-comment since it lies too far outside github's diff settings.

This whole comment is pretty subjective, and it's up to you.

Reply:
Implemented this check in a slightly different way.